Assets/Scripts/Layers/LayerTypes/LayerNL3DBase.cs

```diff
@@ -25,6 +25,7 @@ protected virtual void Awake()
         protected virtual void OnDestroy()
         {
             LayerData.RemoveLayer(this);
+            Destroy(UI.gameObject);
```

Comment:
Misschien beter om het destroyen van de gameObject in LayerUI te doen en van LayerUI een methode aan te roepen om deze te verwijderen. Nu lek je dat er een gameObject is naar deze class en breek je de encapsulatie van de LayerUI class

Reply:
done